### ISSUE TITLE
HTML Library

### ISSUE THREAD

**Issue by bryanjos** [elixirscript/elixirscript#374]
Something similar to how elm-html works

**Comment by vans163** [comment]
seems like this should belong in ecosystem.

**Comment by bryanjos** [comment]
That's the goal. I should make that more clear

**Comment by Harleymckee** [comment]
this include virtual dom implementation like https://github.com/elm-lang/virtual-dom

**Comment by bryanjos** [comment]
@Harleymckee yes that would mostly like be the most difficult part if something was built from scratch. Or a third-party library could be used.

**Comment by OvermindDL1** [comment]
> @Harleymckee yes that would mostly like be the most difficult part if something was built from scratch. Or a third-party library could be used.

Eh, or I could just port my bucklescript-tea VDom over.

**Comment by bryanjos** [comment]
@OvermindDL1 that would be awesome!

**Comment by OvermindDL1** [comment]
> @OvermindDL1 that would be awesome!

It is something that I'd been planning for a while of whenever I get time and ElixirScript finalizes it's API.  ^.^

**Comment by bryanjos** [comment]
Cool! I think things will be pretty stable now. Just count on using Elixir's Standard Library and using the FFI where needed. There may be changes if or when processes are implemented, but those should be underlying.

**Comment by OvermindDL1** [comment]
Is the DOM already FFI'd as a set of interfaces yet?  Or does that need to be manually FFI'd?

**Comment by bryanjos** [comment]
That needs to be manually FFI'd. I haven't made any and I don't think any exists in the eco system yet.

**Comment by OvermindDL1** [comment]
> That needs to be manually FFI'd. I haven't made any and I don't think any exists in the eco system yet.

All fine, something else I can make too unless someone beats me to it.  ^.^

Feel free to ping me on occasion to remind me, when my free time randomly pops up I tend to pick random things unless someone is pinging me about something.  :-)

**Comment by bryanjos** [comment]
Will do

**Comment by bryanjos** [comment]
@OvermindDL1 do you know what DOM FFIs would be needed? I could create something in my free time for that

**Comment by OvermindDL1** [comment]
@bryanjos I'll probably port my bucklescript-tea VDom over (not verbatum, it has a pending redesign) and it exports from the DOM only what it has needed so far, and you can see all of those in the ml files that start with `web` at https://github.com/OvermindDL1/bucklescript-tea/tree/master/src if you want to get those brought over?  That would then be the minimal necessary that I'd need.  :-)

**Comment by bryanjos** [comment]
I've made this so far https://github.com/elixirscript/elixirscript_web

**Comment by bryanjos** [comment]
I went ahead and published it to hex https://hex.pm/packages/elixir_script_web

**Comment by OvermindDL1** [comment]
Ahh the github names and hex names do not match!  ^.^;
Hehe, I love that you reference MDN in the docs, that is just what everyone should do.  :-)

Keep pinging me and as I get time I'll start the porting.  :-)
